src/System.Globalization/tests/PlatformDetection.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Diagnostics;
+using System.IO;
+
+public static class PlatformDetection
+{
+    private static bool? s_isUbuntu1510;
+
+    public static bool IsUbuntu1510
+    {
+        get
+        {
+            if (!s_isUbuntu1510.HasValue)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    IdVersionPair v = ParseOsReleaseFile();
+
+                    if (v.Id == "ubuntu" && v.VersionId == "15.10")
+                    {
+                        s_isUbuntu1510 = true;
+                        return true;
+                    }
+                }
+
+                s_isUbuntu1510 = false;
+            }
+
+            return s_isUbuntu1510.Value;
+        }
+    }
+
+    private static IdVersionPair ParseOsReleaseFile()
+    {
+        Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.Linux));
+
+        IdVersionPair ret = new IdVersionPair();
+
+        foreach (string line in File.ReadLines("/etc/os-release"))
+        {
+            if (line.StartsWith("ID=", System.StringComparison.Ordinal))
+            {
+                ret.Id = line.Substring("ID=".Length);
+            }
+            else if (line.StartsWith("VERSION_ID=", System.StringComparison.Ordinal))
+            {
+                ret.VersionId = line.Substring("VERSION_ID=".Length);
+            }
+        }
+
+        string versionId = ret.VersionId;
+
+        if (versionId != null && versionId.Length >= 2 && versionId[0] == '"' && versionId[versionId.Length - 1] =='"')
+        {
+            // Remove Quotes.
+            ret.VersionId = versionId.Substring(1, versionId.Length - 2);
+        }
+
+        return ret;
```

Comment:
The problem is that OSDescription is basically the output of `uname -mrv` which doesn't really tell you what's going on with the ICU on the system.  I think it makes more sense to parse this file instead of trying to infer kernel versions to distros.  I will guard this parse `/etc/os-release` with an exists check to not fail in the case where the file doesn't exist (e.g. some other distro).